Lib/asyncio/queues.py

```diff
@@ -214,6 +250,32 @@ async def join(self):
         if self._unfinished_tasks > 0:
             await self._finished.wait()
 
+    def shutdown(self, immediate=False):
+        """Shut-down the queue, making queue gets and puts raise QueueShutDown.
+
+        By default, gets will only raise once the queue is empty. Set
+        'immediate' to True to make gets raise immediately instead.
+
+        All blocked callers of put() will be unblocked, and also get()
+        and join() if 'immediate'.
+        """
```

Comment:
Docstring to modify depending of agree/disagree on my first remark about `blocked callers'.
Changes could be:
```
All blocked callers of put() and get() will be unblocked, and
also join() if 'immediate
```